pkg/model/field.go

```diff
@@ -63,6 +63,23 @@ func (f *Field) IsRequired() bool {
 	return util.InStrings(f.Names.ModelOriginal, f.CRD.Ops.Create.InputRef.Shape.Required)
 }
 
+// GetSetterConfig returns the SetFieldConfig object associated with this field
+// and a supplied operation type, or nil if none exists.
+func (f *Field) GetSetterConfig(opType OpType) *ackgenconfig.SetFieldConfig {
+	if f.FieldConfig == nil {
+		return nil
+	}
+	rmMethod := ResourceManagerMethodFromOpType(opType)
+	for _, setCfg := range f.FieldConfig.Set {
+		// If the Method attribute is nil, that means the setter config applies to
+		// all resource manager methods for this field.
+		if setCfg.Method == nil || strings.EqualFold(rmMethod, *setCfg.Method) {
```

Comment:
What should be the strategy if there is a setConfig with empty method name and one setConfig with OpType from parameter?

I think currently a single one will be returned depending on which is defined earlier. Should a priority be added for method match first and then fallback to empty method?